### PR TITLE
Add reset search (X) to toolbar search input

### PR DESCRIPTION
**Pull request by vosco88**
as referenced in #3425 

This is how it would look
image

@LukeTowers feel free to check this (as this is my first bigger PR I do not know if I have taken correct approach by using just the october js framework)
Right now the X is always visible - I was thinking to have it appear only when there is input present but that would not solve the issue with the invisible search that you mentioned.

### CODE REVIEW

**Comment by vosco88** [comment]
@w20k I have not yet commited the recompiled assets as I have some other things there I am testing and would need to clear them first, that might cause it not working properly. I just wanted to share this if I am going the right direction.

**Comment by w20k** [comment]
@vosco88, let me know when I could start testing your PR. Thanks!

**Comment by vosco88** [comment]
@w20k I've cleared my additional data and recompiled storm.css. Hopefully you should be able to test now. Thanks

**Comment by w20k** [comment]
@vosco88, could you show clear button only when search input is not empty? For me, it just stays there 😄

**Comment by vosco88** [comment]
@w20k I was thinking about it, but in the original issue Luke mentioned that sometimes the input does not get filled with the session data and the list is then filtered by an invisible search. In that case the empty input would me no reset search button. So I wanted to wait on what Luke thinks. But for sure I can implement it. Where would be a good place to put the javascript code for this? In which file?

**Comment by w20k** [comment]
@vosco88, If that's the case you'd have to add clear button only when you do a 'search' request.

Not 100% sure, might wait for @LukeTowers on this one, but could be inside `system/assets/ui/js/filters.js`.

**Comment by LukeTowers** [comment]
@vosco88 that's not actually an issue, the issue is that it does get populated however it's not immediately apparent to the user that the search box actually has stuff in it from the session that is affecting the search.

**Comment by vosco88** [comment]
Oh, I understand now. So I might do it differently then. @LukeTowers  any hint on where to put the javascript code for showing the button only when there is an input filled? would it be fine as per suggestion above into the filter.js?

**Comment by LukeTowers** [comment]
@vosco88 you could use the `oninput` HTML attribute on the search box itself to check it's value and hide / show the clear button based on that.

**Comment by vosco88** [comment]
@LukeTowers used that, works ok, only thing that remains to check on page load if there is any input and show the button if there is. `onload` on the input does not work. Where could I place the document onload check for this? Thanks for the help

**Comment by vosco88** [comment]
@LukeTowers why haven't I thought of that? :) ok now it is done

**Comment by w20k** [comment]
@vosco88, could you add make left padding 1px bigger, I can see the search icon: `padding-left: 6px;`. And also remove outline from the button.

```    padding-left: 6px;

**Comment by vosco88** [comment]
@w20k Just done it.

**Comment by w20k** [comment]
@vosco88, looks good!

dec-13-2018 15-49-39

File: modules/backend/widgets/search/partials/_search.htm [hunk 1]
```diff
@@ -3,18 +3,17 @@
...
         data-load-indicator-opaque
         class="form-control <?= $cssClasses ?>"
         autocomplete="off" />
     <button class="clear-input-text" 
             value=""
-			data-request="<?= $this->getEventHandler('onSubmit') ?>"
-			data-load-indicator
-			data-load-indicator-opaque
-			data-request-success="$(this).prev().val('');"
+			style="<?php if($value == '') echo 'display:none' ?>"
```
Comment (LukeTowers) [inline modules/backend/widgets/search/partials/_search.htm]: change to `style="<?= empty($value) ? 'display: none;' : ''; ?>"`

File: modules/backend/widgets/search/partials/_search.htm [hunk 2]
```diff
@@ -3,18 +3,17 @@
         placeholder="<?= $placeholder ?>"
         type="text"
         name="<?= $this->getName() ?>"
-        value="<?= $value ?>"
+        value="<?= $value ?>"		
         data-request="<?= $this->getEventHandler('onSubmit') ?>"
+		data-request-complete="if($(this).val().length) {$(this).next().show()} else {$(this).next().hide()};"
```
Comment (LukeTowers) [inline modules/backend/widgets/search/partials/_search.htm]: Use spaces, not tabs - and please remove trailing whitespace

File: modules/backend/widgets/search/partials/_search.htm (hunk 1 above)
Comment (vosco88) [reply modules/backend/widgets/search/partials/_search.htm]: done

File: modules/backend/widgets/search/partials/_search.htm (hunk 2 above)
Comment (vosco88) [reply modules/backend/widgets/search/partials/_search.htm]: hopefully also done

File: modules/system/assets/ui/less/toolbar.less [hunk 3]
```diff
@@ -27,6 +27,23 @@
...
+				top: 2px;
+				background-color: #fff;
+				padding-left:6px;
+				height:30px;
+				border: 0;
+			}
+			
+		.clear-input-text:focus {
+			outline:none;
+		}
```
Comment (LukeTowers) [inline modules/system/assets/ui/less/toolbar.less]: Please fix the indentation on these styles, should be four spaces and inline with the others in the same nested context

File: modules/backend/widgets/search/partials/_search.htm [hunk 4]
```diff
@@ -3,11 +3,17 @@
         placeholder="<?= $placeholder ?>"
         type="text"
         name="<?= $this->getName() ?>"
-        value="<?= $value ?>"
+        value="<?= $value ?>"
```
Comment (LukeTowers) [inline modules/backend/widgets/search/partials/_search.htm]: Still has trailing whitespace

File: modules/backend/widgets/search/partials/_search.htm [hunk 5]
```diff
@@ -3,11 +3,17 @@
...
         <?= !$searchOnEnter ? 'data-track-input' : '' ?>
         data-load-indicator
         data-load-indicator-opaque
         class="form-control <?= $cssClasses ?>"
         autocomplete="off" />
+    <button class="clear-input-text" 
+            value=""
+			style="<?= empty($value) ? 'display: none;' : ''; ?>"
+			onclick="$(this).prev().val('');$(this).prev().request();"
+		><i class="icon-times"></i></button>
```
Comment (LukeTowers) [inline modules/backend/widgets/search/partials/_search.htm]: Please change to 
```html
<button
    class="clear-input-text"
    value=""
    style="<?= empty($value) ? 'display: none;' : ''; ?>"
    onclick="$(this).prev().val('');$(this).prev().request();"
>
    <i class="icon-times"></i>
</button>
```

**Comment by vosco88** [comment]
@LukeTowers I hope it should be fine now, Sorry for so many changes needed.

**Pull request merged**